### PR TITLE
Remove `out_type` from 1D core, perform formatting in the frontends

### PR DESCRIPTION
**Pull request by lostanlen**
Address (but does not fully fix)  #852 

Third PR in a streak (after #860 and #861), attempting to simplify core 1D scattering and decouple its functionalities.
This time the offender is `out_type` and the post-processing that comes after the scattering transform per se.

It turns out that there isn't much of a reason to write a backend-agnostic implementation of this formatting stage, considering that it's already divided into `"array"`, `"dict"` and `"list"` cases.

Besides, this PR brings us closer than we've ever been to a `Scattering1D` generator, which would `yield` coefficients instead of `return`-ing them. Hence a _big_ simplification in scope once we're able to decouple `scattering1d` from padding, averaging, and unpadding routines.

### CODE REVIEW

**Comment by MuawizChaudhary** [comment]
I get the impression that we have to rebase this on top of #861 ?

**Comment by lostanlen** [comment]
> I get the impression that we have to rebase this on top of #861 ?

Yes. #861 takes precedence

File: kymatio/scattering1d/frontend/numpy_frontend.py [hunk 1]
```diff
@@ -27,26 +27,26 @@ def scattering(self, x):
...
             for k, v in S.items():
                 # NOTE: Have to get the shape for each one since we may have
                 # average == False.
                 scattering_shape = v.shape[-2:]
                 new_shape = batch_shape + scattering_shape
-
                 S[k] = v.reshape(new_shape)
         elif self.out_type == 'list':
             for x in S:
+                x.pop('n')
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/numpy_frontend.py:47]: Why do we pop in the first place?

**Review by MuawizChaudhary — APPROVED** [review]
I approve these changes. @lostanlen I'll merge once you've addressed my comment.

File: kymatio/scattering1d/frontend/numpy_frontend.py (hunk 1 above)
Comment (lostanlen) [reply kymatio/scattering1d/frontend/numpy_frontend.py:47]: hm ... you're right, we don't have to. I think that we did this to match the returned formats of core 2D. Note that the dictionaries in the `out_S` list do not have an `n` field.
https://github.com/kymatio/kymatio/blob/master/kymatio/scattering2d/core/scattering2d.py

I can't see another justification for what we're doing now in 1D, i.e. to redact metadata that could potentially be useful for the user. `n`can be seen as a "pointer" (or tuple of pointers) to the filterbank `psi1_f` (or tuple of filterbanks `(psi1_f, psi2_f)`)

i'd be in favor of not popping it away. that change is backwards compatible and very easy to maintain (we'd be _removing_ a for loop!) This change adds some new functionality; but it's one that the users already had access too in a different way, namely, `S.keys()`under `out_type="dict"`

TL;DR i can delete that line if that makes you happy!

Comment (MuawizChaudhary) [reply kymatio/scattering1d/frontend/numpy_frontend.py:47]: Lets do it in a separate repo. For now can you create an issue?

**Comment by MuawizChaudhary** [comment]
I'm squashing and merging. @lostanlen pls create an issue as discussed above.
